### ISSUE TITLE
Commit conflict when running compaction on quiet dataset

### ISSUE THREAD

**Issue by westonpace** [lancedb/lance#1836]
From Discord:

```
I'm getting an error when I compact my dataset:

OSError: Commit conflict for version 607: Failed to commit the transaction after 5 retries., /home/runner/work/lance/lance/rust/lance/src/io/commit.rs:399:19

I've now rebuilt the whole thing from scratch a few times now (~1TB in size) and I'm still getting the same error. Unfortunately due to size it's hard for me give a bug that reproduces well
```

The user is creating the dataset in parallel with many writes.  They then copy the dataset to a new location and attempt to run commit but get the above error.

**Comment by wjones127** [comment]
I believe this is a duplicate of #2397
